recipes/uvw/config.yml

```
@@ -0,0 +1,3 @@
+versions:
+  "2.6.0":
+    folder: "2.6.x"
```

```suggestion
    folder: "all"
```

Is there any restriction about the version? Otherwise, use `all`.

---

Each minor version requires specific version of libuv (see https://github.com/skypjack/uvw/releases). 2.6.x requires libuv 1.38 (see https://github.com/conan-io/conan-center-index/blob/57d549f892647d08368c267fdb58b25c39c4f3ed/recipes/uvw/2.6.x/conanfile.py#L23
) (and 2.7.x will probably require libuv 1.39)

So we need to have a recipe for each minor version.

---

no problem, you can use a dict to filter it:

```python
def requirements(self):
    if tools.Version(self.version) ~= "2.6":
        self.requires("libuv/1.38.1")
    elif tools.Version(self.version) ~= "2.7":
        self.requires("libuv/1.39.0")    
```

---

I think the best solution would be to add a new field "requirements" in the conandata.yml. Unfortunately, [KB-H030](https://github.com/conan-io/conan-center-index/wiki/Error-Knowledge-Base#kb-h030-conandatayml-format) and [KB-H031](https://github.com/conan-io/conan-center-index/wiki/Error-Knowledge-Base#kb-h031-conandatayml-reduce) doesn't allow it. 

But I think allowing it would encourage bad practices and misuses.

---

It won't happen, because as you said "it would encourage bad practices and misuses." Which is true, but also this is particular case, changing hooks and conandata.yml because a specific case is not correct.

---

again, use the condition based on the version, that's a regular approach:

https://github.com/conan-io/conan-center-index/blob/67b15f79611c10a04d2207763a75a9a863018f08/recipes/folly/all/conanfile.py#L78
https://github.com/conan-io/conan-center-index/blob/305618b987953292cedf3dd5fc7e0c9401699b28/recipes/spdlog/all/conanfile.py#L51